### ISSUE TITLE
Tisserand plots

### ISSUE THREAD

**Issue by jorgepiloto** [poliastro/poliastro#925]
📕 **About Tisserand plots**

Imagine you are observing asteroids which are orbiting near Jupiter. The big
planet might create perturbations among these small bodies orbits but you want
to make sure if an actual observed body was previously observed, althought their
COE parameters might be suffered some variations.

Previous question arised to Félix Tisserand, who was a French astronomer that in
the 19th century developed a method based on an invariant constant (which today
carries his name). You can understand the tisserand parameters as some way to
measure the "energy" of a body. In fact, its units are *distance2 / time2* which
is also known as "characteristic energy".

Nowadays, Tisserand graphs can be plotted in many different ways:

* Tisserand parameter VS apoapsis || eccentricity || inclination
* Heliocentric periapsis VS Heliocentric apoapsis
* Heliocentric orbital period VS Heliocentric apoapsis

The most common ones are the last two, though the first one might be plotted by
making use of a contourf plot as seen in this page:

* [All Tisserand Solutions by contourf plots](https://armchairastronautics.blogspot.com/search?q=tisserand)

The most powerful feature this kind of plots show is related to mission design,
in particular to gravity assist maneuvers. They would lead to the solution to
the question: if I want to go from body A to body B, what would be the most
optimal set of flybys to achieve the trip?

![poliastro_tiss](https://user-images.githubusercontent.com/28702884/79787422-40b8f000-8347-11ea-8285-d8d8cba2fe39.png)

💡 **Feature request**

The algorithm on how to obtain these kind of plots is very well explained in the
following [document](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=22&ved=2ahUKEwje9d2ZzvfoAhUQLBoKHeYXAbMQFjAVegQIBhAB&url=https%3A%2F%2Ffenix.tecnico.ulisboa.pt%2FdownloadFile%2F1407770020545591%2FMSc_DavidPalma_Final_13Dec.pdf&usg=AOvVaw2-9skb6UG2tix-UieDnAFT). Furthermore, it includes lots of solved cases which could be used as unit tests.

With the implementation of the function `get_mean_elements()` it is now possible
to obtain the necessary parameters for any desired body.

🎯 **Goal**

Although the working algorithm is pretty clear, the problem might arise on how
to implement this feature inside *poliastro*. I am thinking about the following
usage:

```python
# File poliastro/plotting/tisserand.py

class Tisserand:
    """ Plots a Tisserand parameter contourf """

    def __init__(a_span, ecc_span, icc_span):
        ...

class FlybyPlotter():
    """ Plots Tisserand graphs for given bodies and V_inf span """

    def plot(body, Vinf_span, **kargs)
```

Previous code might lead to the following working flow:

```python
from poliastro.plotting.tisserand import FlybyPlotter
from poliastro.bodies import Venus, Earth, Jupiter
import numpy as np

fp = FlybyPlotter()
fp.plot(Venus, np.linspace(0, 10, 11) * u.km / u.s)
fp.plot(Earth, np.linspace(0, 9, 10) * u.km / u.s)
fp.plot(Jupiter, np.linspace(0, 15, 16) * u.km / u.s)
```

I would like to read your opinions and suggestions on how to adapt the algorithm
to the actual API, not pretty happy with the previous "FlybyPlotter", migh lead
to confusion.

🔖 **References and papers**

* [Preliminary Trajectory Design of a Mission to Enceladus](https://www.researchgate.net/publication/318043268_Preliminary_trajectory_design_of_a_mission_to_Enceladus)

* [Graphical method for Gravity-assist sequancing](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=15&cad=rja&uact=8&ved=2ahUKEwiqmLXu0vfoAhVQLBoKHc-VA9IQFjAOegQIAhAB&url=https%3A%2F%2Fengineering.purdue.edu%2Fpeople%2Fjames.m.longuski.1%2FJournalArticles%2F2002%2FGraphicalMethodforGravity-AssistTrajectoryDesign.pdf&usg=AOvVaw3ZPhmBjUII1W1HtkOlVnVV)

**Comment by astrojuanlu** [comment]
Awesome @jorgepiloto, thanks for adding all the technical background :heart: 

Perhaps we can do something like this?

```python
from poliastro.plotting.tisserand import FlybyPlotter
from poliastro.bodies import Venus, Earth, Jupiter

fp = TisserandPlotter()  # defaults to 10 contours
fp.plot(Venus, (0, 10) * u.km / u.s, num_contours=11)
fp.plot(Earth, (0, 9) * u.km / u.s,  # no need to specify num_contours=10
fp.plot(Jupiter, (0, 15) * u.km / u.s, 16)
```

(I renamed the object to `TisserandPlotter` and removed the need to import `np.linspace` by giving a pair (start, end)).

Let's save the FlyByPlotter for #662 :smirk:

**Comment by astrojuanlu** [comment]
Closed in #927 :tada:
